velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  explicit LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar)), RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
```

Comment:
Do we also need to check that every time?
I thought that we have some safeguards during the expr compilation or at some earlier stage and here we are pretty much guaranteed to have correct arguments.

Reply:
Not really necessary, but at least a DCHECK is fine IMO.

However LIKE should have exactly two arguments.

Reply:
I was following the style consistent with the rest of the functions in this file.

@mbasmanova : Please can you confirm.

Reply:
I don't think it is a big issue, but I agree that these checks are redundant. We could enhance the expression evaluation framework to perform these checks using signatures specified by the function.

Reply:
@funrollloops 

> However LIKE should have exactly two arguments.

In Presto like takes 2 or 3 arguments: https://prestodb.io/docs/current/functions/comparison.html#like